### PR TITLE
Validation refinements informed by Cold Exposure data 2 - Strip units

### PR DESCRIPTION
**Pull request by hepcat72**
# Please review #620 first - DONE/MERGED

## Summary Change Description

A simple improvement inspired by the Cold Exposure data processing.

## Affected Issue Numbers

- Resolves a ToDo item in the comments of issue #580

## Code Review Notes

Added a method called strip_units to the accucor loading code.  If the type isn't string, it just returns the submitted value, assuming that pandas successfully converted the value and that units stripping is not needed.  It adds affected field values to an array and issues a warning at the end of the script that includes the column and a list of rows.

## Checklist

- [ ] Changes have been added to the *Unreleased* section in the [`changelog.md`](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md).
  - Will add during merge. Branched before the CHANGES.md file was added.
- [x] All issue requirements satisfied (or no linked issues)
- [x] [Linting passes](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting).
- [x] [Migrations created & committed *(or no model changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)
- Tests
  - [x] [Tests implemented *(or no code changes)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All example load tests pass (remotely) *(or their failures predated and are unrelated to this branch)*

### CODE REVIEW

**Comment by hepcat72** [comment]
I'll address the conflict next week and add in-line code review notes.

**Comment by hepcat72** [comment]
Resolved conflicts in a test branch.  Waiting for tests to pass to make sure everything's OK.  Will merge back to this branch if they all pass.

**Comment by hepcat72** [comment]
Not sure why the tests and linting are rerunning.  I haven't pushed any changes to this branch (yet)...

**Comment by hepcat72** [comment]
@lparsons @fkang-pu @jcmatese - This PR and #623 are both ready for review.

**Review by fkang-pu — APPROVED** [review]
Nice improvement by adding "strip_units" method.

**Review by jcmatese — APPROVED** [review]

File: DataRepo/utils/sample_table_loader.py [hunk 1]
```diff
@@ -490,15 +503,58 @@ def get_treatment(self, rownum, row):
...
+        if type(val) != str:
+            # Assume that if it's not a string, it already doesn't contain units, because pandas converted it
+            return val
+
+        stripped_val = val
+
+        united_val_pattern = re.compile(
+            r"^(?P<val>[\d\.eE]+)\s*(?P<units>[a-zA-Z][a-zA-Z\/]*)$"
+        )
+        match = re.search(united_val_pattern, val)
```
Comment (lparsons) [inline DataRepo/utils/sample_table_loader.py:521]: Perhaps just a simple call to `[split()](https://www.w3schools.com/python/ref_string_split.asp)` here? Something like `val.split(' ', 1)`.

**Review by lparsons — APPROVED** [review]
I'm a bit skeptical of munging data like this, but I can see how it could be helpful if users add units frequently.

**Comment by hepcat72** [comment]
Merging.  If you like the units error suggestion I made @lparsons, I'll create an issue.

**Comment by lparsons** [comment]
Adding an issue is fine, but I don't want to take the focus off of getting mzXML files stored.

To date, we have only processed seventeen studies. Eventually I think enhancements like this could be quite valuable, but right now small time savings on each is not a priority. Adding complexity to various components means that the inevitable refactors we will need to do to add new types of data, etc. can take longer. I would prefer to work on the must have features when the codebase and feature set is lean to avoid having to refactor increasingly complex code.

**Pull request merged**